clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -176,20 +176,32 @@ bool
 check_branch_target_after_branch()
 {
     std::cerr << "Testing branch targets\n";
+    constexpr memref_tid_t TID = 1;
+    constexpr uintptr_t TIMESTAMP = 3;
     // Correct simple test.
     {
         std::vector<memref_t> memrefs = {
-            gen_instr(1, 1), gen_branch(1, 2),
-            gen_instr(1, 3), gen_marker(2, TRACE_MARKER_TYPE_TIMESTAMP, 0),
-            gen_instr(2, 1),
+            gen_instr(TID, 1),     gen_branch(TID, 2),
+            gen_instr(TID, 3),     gen_marker(TID + 1, TRACE_MARKER_TYPE_TIMESTAMP, 0),
+            gen_instr(TID + 1, 1),
+        };
+        if (!run_checker(memrefs, false))
+            return false;
+    }
+    // Correctly skip CPU ID marker.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_instr(TID, 1),
+            gen_branch(TID, 2),
+            gen_marker(TID, TRACE_MARKER_TYPE_TIMESTAMP, TIMESTAMP),
```

Comment:
Wait: the branch is supposed to be delayed across the timestamp+cpuid.  So this shouldn't happen.  We have raw2trace unit tests of this type of thing -- so is there some corner case where you saw this in the wild?

Reply:
The corner case is the rollback after rseq side exit. I've updated to the PR accordingly.